src/Api/PubnubApi/Builder/UriUtil.cs

```diff
@@ -103,7 +103,7 @@ public static string EncodeUriComponent(bool forPamSign, string s, PNOperationTy
             }
 
             encodedUri = o.ToString();
-            if (type == PNOperationType.PNHereNowOperation || type == PNOperationType.PNHistoryOperation || type == PNOperationType.Leave || type == PNOperationType.PNHeartbeatOperation || type == PNOperationType.PushRegister || type == PNOperationType.PushRemove || type == PNOperationType.PushGet || type == PNOperationType.PushUnregister)
+            if (type == PNOperationType.PNHereNowOperation || type == PNOperationType.PNHistoryOperation || type == PNOperationType.PNFetchHistoryOperation || type == PNOperationType.Leave || type == PNOperationType.PNHeartbeatOperation || type == PNOperationType.PushRegister || type == PNOperationType.PushRemove || type == PNOperationType.PushGet || type == PNOperationType.PushUnregister)
```

Comment:
This is getting kind of crazy, we should refactor it some other time.

Reply:
will refactor for this version for review

Reply:
code refactored. please review.